src/marker.rs

```diff
@@ -357,6 +357,114 @@ fn parse_line_list_msg(
     lines
 }
 
+fn parse_sphere_msg(
+    msg: &rosrust_msg::visualization_msgs::Marker,
+    color: &tui::style::Color,
+    iso: &Isometry3<f64>,
+) -> Vec<Line> {
+    let mut lines: Vec<Line> = Vec::new();
+
+    let scale = &Point3::new(msg.scale.x, msg.scale.y, msg.scale.z);
+
+    let p1 = iso.transform_point(&Point3::new(-scale.x * 0.5, 0.0, 0.0));
+    let p2 = iso.transform_point(&Point3::new(scale.x * 0.5, 0.0, 0.0));
+    let p3 = iso.transform_point(&Point3::new(0.0, -scale.y * 0.5, 0.0));
+    let p4 = iso.transform_point(&Point3::new(0.0, scale.y * 0.5, 0.0));
+    let p5 = iso.transform_point(&Point3::new(0.0, 0.0, -scale.z * 0.5));
+    let p6 = iso.transform_point(&Point3::new(0.0, 0.0, scale.z * 0.5));
+
+    //central "cross" showing the main axes and the center
+    lines.push(Line {
+        x1: p1.x,
+        y1: p1.y,
+        x2: p2.x,
+        y2: p2.y,
+        color: *color,
+    });
+    lines.push(Line {
+        x1: p3.x,
+        y1: p3.y,
+        x2: p4.x,
+        y2: p4.y,
+        color: *color,
+    });
+    lines.push(Line {
+        x1: p5.x,
+        y1: p5.y,
+        x2: p6.x,
+        y2: p6.y,
+        color: *color,
+    });
+
+    let segment_count = 20; //number of segments of each ellipse for a pair of axes
+    let step = (2.0 * PI) / (segment_count as f64); //the angle enclosing one segment if the axes were of equal sizes
```

Comment:
nit pick: The step doesn't need explanation.  The line is also very long and wraps when when i have to files open next to each other, i hate that :wink:

Reply:
Done. Though, I have to say you guys are really fussy 🙄.